File: Server/src/main/java/server/ReservationService.java
package server;

import model.domain.Reservation;
import model.domain.Trip;
import model.utils.MyException;
import repository.IReservationRepo;

import java.util.Collection;

public class ReservationService {
    private final IReservationRepo repo;

    public ReservationService(IReservationRepo repo) {
        this.repo = repo;
    }

    public void saveReservation(Reservation reservation) throws MyException { repo.save(reservation); }

    public Collection<Reservation> getAllReservations() { return repo.getAll(); }

    public int getAvailableSeatsForTrip(Trip trip) { return repo.getAvailableSeatsForTrip(trip); }
}

File: Server/server/ReservationService.cs
using System;
using System.Collections.Generic;
using Model;
using Persistence;

namespace Server.server
{
    public class ReservationService
    {
        private readonly IReservationRepo _repo;

        public ReservationService(IReservationRepo repo)
        {
            _repo = repo;
        }

        public void SaveReservation(Reservation reservation) =>
            _repo.Save(reservation);

        public IEnumerable<Reservation> GetAllReservations() => _repo.GetAll();

        public int GetAvailableSeatsForTrip(Trip trip) => _repo.getAvailableSeatsForTrip(trip);
    }
}
